logger.js

```diff
@@ -82,45 +95,45 @@ Logger.prototype.destroy = function destroy() {
     }, this);
 };
 
-module.exports = Logger;
-
-function logMethod(logger, levelName, level, path) {
-    var streams = logger.streams;
-    var logStreams = level.backends.reduce(function (acc, name) {
-        if (!streams[name]) {
-            return acc;
-        }
-
-        acc[name] = streams[name];
-        return acc;
-    }, {});
+Logger.prototype.log = function log(entry, callback) {
+    var levelName = entry.level;
+    var level = this.levels[levelName];
+    var logStreams = this.streamsByLevel[levelName];
+    var logger = this;
+    if (this.statsd && typeof this.statsd.increment === 'function') {
+        this.statsd.increment('logtron.logged.' + levelName);
+    }
 
-    return log;
+    level.transforms.forEach(function (transform) {
+        entry = transform(entry);
+    });
 
-    function log(message, meta, callback) {
-        var entry = new Entry(levelName, message, meta, path);
+    parallelWrite(logStreams, entry, function (err) {
+        if (!err) {
+            if (callback) {
+                callback(null);
+            }
+            return;
+        }
 
-        if (this.statsd && typeof this.statsd.increment === 'function') {
-            this.statsd.increment('logtron.logged.' + levelName);
+        if (callback && typeof callback === 'function') {
+            return callback(err);
         }
 
-        level.transforms.forEach(function (transform) {
-            entry = transform(entry);
-        });
+        logger.emit('error', err);
+    });
+};
 
-        parallelWrite(logStreams, entry, function (err) {
-            if (!err) {
-                if (callback) {
-                    callback(null);
-                }
-                return;
-            }
+Logger.prototype.createChild = function (subPath, levels) {
+    return new this.ChildLogger({
+        target: this,
+        path: subPath,
+        levels: levels
+    });
+};
 
-            if (callback && typeof callback === 'function') {
-                return callback(err);
-            }
+Logger.prototype.makeLogMethod = ChildLogger.prototype.makeLogMethod;
```

Comment:
no it does not. you can put the floating function declaration in a seperate file shared by both.